### ISSUE TITLE
quantum_router needlessly hits the admin API

### ISSUE THREAD

**Issue by dwcramer** [ansible/ansible-modules-core#2113]
I'm using the quantum_router module to create a router in my tenant, but it fails when it tries to get a list of tenants using the admin API, which I don't have access to since I'm just a user of a single tenant. Why is it necessary to get a list of tenants? I've provided the tenant name and also have OS_TENANT_ID, etc set. I suspect the same problem exists for quantum_router_gateway and quantum_router_interface. 

If I'm lucky, @emonty will notice this ticket.

```
failed: [localhost] => {"failed": true, "parsed": false}
Traceback (most recent call last):
  File "/Users/davcrame/.ansible/tmp/ansible-tmp-1442628676.04-39207445937304/quantum_router", line 1916, in <module>
    main()
  File "/Users/davcrame/.ansible/tmp/ansible-tmp-1442628676.04-39207445937304/quantum_router", line 188, in main
    _set_tenant_id(module)
  File "/Users/davcrame/.ansible/tmp/ansible-tmp-1442628676.04-39207445937304/quantum_router", line 137, in _set_tenant_id
    for tenant in _os_keystone.tenants.list():
  File "/Users/davcrame/dsx/ansible-tools-setup/venv/lib/python2.7/site-packages/keystoneclient/v2_0/tenants.py", line 123, in list
    tenant_list = self._list('/tenants%s' % query, 'tenants')
  File "/Users/davcrame/dsx/ansible-tools-setup/venv/lib/python2.7/site-packages/keystoneclient/base.py", line 113, in _list
    resp, body = self.client.get(url, **kwargs)
  File "/Users/davcrame/dsx/ansible-tools-setup/venv/lib/python2.7/site-packages/keystoneclient/adapter.py", line 170, in get
    return self.request(url, 'GET', **kwargs)
  File "/Users/davcrame/dsx/ansible-tools-setup/venv/lib/python2.7/site-packages/keystoneclient/adapter.py", line 206, in request
    resp = super(LegacyJsonAdapter, self).request(*args, **kwargs)
  File "/Users/davcrame/dsx/ansible-tools-setup/venv/lib/python2.7/site-packages/keystoneclient/adapter.py", line 95, in request
    return self.session.request(url, method, **kwargs)
  File "/Users/davcrame/dsx/ansible-tools-setup/venv/lib/python2.7/site-packages/keystoneclient/utils.py", line 336, in inner
    return func(*args, **kwargs)
  File "/Users/davcrame/dsx/ansible-tools-setup/venv/lib/python2.7/site-packages/keystoneclient/session.py", line 382, in request
    resp = send(**kwargs)
  File "/Users/davcrame/dsx/ansible-tools-setup/venv/lib/python2.7/site-packages/keystoneclient/session.py", line 426, in _send_request
    raise exceptions.ConnectionRefused(msg)
keystoneclient.openstack.common.apiclient.exceptions.ConnectionRefused: Unable to establish connection to https://int.us-texas-3.cloud.cisco.com:35357/v2.0/tenants
```

**Comment by emonty** [comment]
Yo! Yeah - the old modules are hella broken. So ansible/ansible-modules-core#1052 should make you happy. @Shrews and I are going to try to get those landed next week. Also, can we get an account on a metacloud/cisco something? I keep getting bug reports from metacloud/cisco folks and it's hard to fully debug to see if there is something we're not accounting for ...

**Comment by dwcramer** [comment]
That looks promising! Are os_router and os_router_gateway coming as well? Getting very close to my DevOps dream of starting with an empty tenant and setting up **everything** with one idempotent playbook.

I'll check with Niki Acosta about the possibility of getting you a CCS account. For the tenants I've worked with, the OpenStack api & horizon are only accessible on the Cisco network (likewise to access the instances over ssh you have to be on Cisco's network). But perhaps there's some other option I don't know about.

**Comment by emonty** [comment]
On 09/19/2015 04:19 PM, David Cramer wrote:

> That looks promising! Are os_router and os_router_gateway coming as
> well? Getting very close to my DevOps dream of starting with an empty
> tenant and setting up _everything_ with one idempotent playbook.

Yes indeed. I want them landed before 2.0 hits...

> I'll check with Niki Acosta about the possibility of getting you a CCS
> account. For the tenants I've worked with, the OpenStack api & horizon
> are only accessible on the Cisco network (likewise to access the
> instances over ssh you have to be on Cisco's network). But perhaps
> there's some other option I don't know about.

Cool - I emailed her a little while ago about this. Maybe a nudge from 
you would help.

It would be awfully sad if Cisco didn't expose and OpenStack API from 
CCS ...

Monty

**Comment by jctanner** [comment]
@dwcramer the quantum_router module is deprecated and no longer maintained.
